core/src/price_finding/optimization_price_finder.rs

```diff
@@ -193,28 +193,59 @@ pub struct OptimisationPriceFinder {
     io_methods: Arc<dyn Io + Send + Sync>,
     fee: Option<Fee>,
     solver_type: SolverType,
-    price_oracle: Box<dyn PriceEstimating + Sync>,
-    min_avg_fee_per_order: u128,
+    price_oracle: Box<dyn PriceEstimating + Send + Sync>,
+    gas_station: Arc<dyn GasPriceEstimating + Send + Sync>,
+    min_avg_fee_subsidy_factor: f64,
+    default_min_avg_fee_per_order: u128,
```

Comment:
Given we need all these components, would it make sense to create a new "MinAvgFeeComputer"? I find it also slightly weird that the PriceFinder needs to know about gas estimation thingies.

Reply:
I like this idea.